conjur/util/credentials_from_file.py

```diff
@@ -89,19 +90,28 @@ def update_api_key_entry(self, user_to_update, credential_data, new_api_key,
         hosts[credential_data['machine']] = (user_to_update, None, new_api_key)
         self.build_netrc(netrc_obj)
 
-    def remove_credentials(self, conjurrc_conjur_url):
+    def remove_credentials(self, conjurrc):
         """
         Method that removes the described login entry from netrc
         """
-        netrc_data = self.load(conjurrc_conjur_url)
+        logging.debug(f"Attempting to remove credentials from '{DEFAULT_NETRC_FILE}'")
+        # pylint: disable=no-else-return
+        if not os.path.exists(DEFAULT_NETRC_FILE):
+            return
+        elif os.path.exists(DEFAULT_NETRC_FILE) and os.path.getsize(DEFAULT_NETRC_FILE) != 0:
+            netrc_data = self.load(conjurrc.conjur_url)
+
+            netrc_obj = netrc.netrc(DEFAULT_NETRC_FILE)
+            hosts = netrc_obj.hosts
+            hosts.pop(netrc_data['machine'], None)
 
-        netrc_obj = netrc.netrc(DEFAULT_NETRC_FILE)
-        hosts = netrc_obj.hosts
-        hosts.pop(netrc_data['machine'], None)
+            netrc_obj = netrc.netrc(DEFAULT_NETRC_FILE)
+            netrc_obj.hosts.pop(netrc_data['machine'], None)
+            self.build_netrc(netrc_obj)
+        else:
+            raise Exception("You are already logged out.")
```

Comment:
this exception can be thrown if file is not empty but machine not in it right?

Reply:
or this is a bug if this scenario happens

Reply:
as discussed load is called which handles this